### PR TITLE
Fix multi-line array indentation.

### LINKED ISSUE #33: Array indentation seems broken
Multiline array literals get formatted like this:

```
  $nodes = [
            '10.116.1.72',
            '10.116.1.73'
            ]
```

which seems quite broken. I think they should look like this:

```
  $nodes = [
    '10.116.1.72',
    '10.116.1.73'
  ]
```

### PR DESCRIPTION
**Pull request by askreet**
This fixes #33 - I know you mentioned wanting to implement SMIE instead of patch the existing implementation. Unfortunately, I didn't see that issue until after I fixed it locally. I understand if you'd prefer to just close this PR in favor of SMIE.

### CODE REVIEW

File: puppet-mode.el [hunk 1]
```diff
@@ -555,11 +555,36 @@ of the initial include plus puppet-include-indent."
...
-          (forward-char -1)
-          (setq cur-indent (current-column))))
+          (if (looking-at "^\\s-*],*")
+              ;; If a closing bracket is on a line by itself, align it with the
+              ;; opening bracket.
+              (progn
+                (goto-char array-start)
+                (setq cur-indent (current-indentation))
+                )
+            (progn
```
Comment (lunaryorn) [inline puppet-mode.el]: You need no `progn` for the else clause.  Please remove it.

File: puppet-mode.el [hunk 2]
```diff
@@ -555,11 +555,36 @@ of the initial include plus puppet-include-indent."
...
+              ;;     'foo',
+              ;;     'bar',
+              ;;   ]
+              ;; }
+              ;;
+              (if (eolp)
+                  (progn
+                    (setq cur-indent (+ (current-indentation) puppet-indent-level)))
+                ;; Otherwise, attempt to align as described above.
+                (progn
```
Comment (lunaryorn) [inline puppet-mode.el]: Likewise here.

File: puppet-mode.el [hunk 3]
```diff
@@ -555,11 +555,36 @@ of the initial include plus puppet-include-indent."
...
-          (re-search-forward "\\S-")
-          (forward-char -1)
-          (setq cur-indent (current-column))))
+          (if (looking-at "^\\s-*],*")
+              ;; If a closing bracket is on a line by itself, align it with the
+              ;; opening bracket.
+              (progn
+                (goto-char array-start)
+                (setq cur-indent (current-indentation))
+                )
```
Comment (lunaryorn) [inline puppet-mode.el]: Please don't keep parenthesis on a separate line.  That's ugly.

**Comment by lunaryorn** [comment]
@askreet Sure, I'll merge it, but please address my remarks.  And please write a comment when you've done so, because Github doesn't send notifications for updated pull requests, so I'd miss your update otherwise.  Thanks :)

**Comment by askreet** [comment]
@lunaryorn Feedback has been addressed. Thanks for the information, this is the most complex elisp hacking I've done to date, so it was helpful.

**Comment by lunaryorn** [comment]
@askreet Merged, thank you!

**Pull request merged**